care/utils/evaluators/base.py

```diff
@@ -0,0 +1,277 @@
+        Subclasses can override to add or modify handlers.
+        """
+        return {
+            "equality": self._handle_equality,
+            "range": self._handle_range,
+            "in_or_equality": self._handle_in_or_equality,
+            "in": self._handle_in,
+            "intersects_any": self._handle_intersects_any,
+            "intersects_all": self._handle_intersects_all,
+        }
+
+    @property
+    def validator_map(self) -> dict[str, Callable[[Any], bool]]:
+        """
+        Mapping of validator names to their functions.
+        Subclasses can override to add or modify validators.
+        """
+        return {
+            "equality": lambda expected_value: isinstance(
+                expected_value, (str, int, float)
+            ),
+            "range": lambda expected_value: isinstance(expected_value, dict)
+            and all(
+                expected_value.get(key) is None
+                or isinstance(expected_value.get(key), (int, float))
+                for key in ["min", "max"]
+                if key in expected_value
+            ),
+            "in_or_equality": lambda expected_value: isinstance(
+                expected_value, (str, list)
+            ),
+            "in": lambda expected_value: isinstance(expected_value, list),
+            "intersects_any": lambda expected_value: isinstance(
+                expected_value, (str, list)
+            ),
+            "intersects_all": lambda expected_value: isinstance(
+                expected_value, (str, list)
+            ),
+        }
+
+    @property
+    def condition_map(self) -> dict[str, str]:
+        """
+        Mapping of condition keys to handler names.
+        Subclasses should override to define their specific conditions.
+        """
+        return {}
+
+    def _handle_equality(self, actual_value: Any, expected_value: Any) -> bool:
+        """
+        Check if actual_value equals expected_value.
+
+        Args:
+            actual_value (Any): The value from the context.
+            expected_value (Any): The expected value for comparison.
+
+        Returns:
+            bool: True if values are equal, False otherwise.
+        """
+        return actual_value == expected_value
+
+    def _handle_range(self, actual_value: Any, expected_value: dict) -> bool:
+        """
+        Check if actual_value falls within min-max range.
+
+        Args:
+            actual_value (Any): The value from the context (expected to be numeric).
+            expected_value (dict): Dictionary with 'min' and/or 'max' keys for range bounds.
+
+        Returns:
+            bool: True if value is within the range, False otherwise.
+        """
+        min_v = expected_value.get("min", float("-inf"))
+        max_v = expected_value.get("max", float("inf"))
+        return min_v <= actual_value <= max_v
+
+    def _handle_in_or_equality(self, actual_value: Any, expected_value: Any) -> bool:
```

Comment:
```suggestion
    def _handle_in_or_equality(self, actual_value: str | int, expected_value: list[str|int] | str | int) -> bool:
```

validate other types in this class as well, using Any doesn't help.